app/models/event.js

```diff
@@ -100,14 +100,14 @@ export default Model.extend({
    *
    * @see app/models/discount-code.js
    */
-  discountCode: belongsTo('discount-code'),
+  discountCode: belongsTo('discount-code', {inverse:null}),
```

Comment:
Lets keep event model clean and specify inverses on the discount-code model instead. Also, specify the correct inverses. Not null. 

```js
  event   : belongsTo('event', {
    inverse: 'discountCodes'
  }), // The event that this discount code belongs to [Form (2)]
  events: hasMany('event', {
    inverse: 'discountCode'
  })    // The events that this discount code has been applied to [Form (1)]
```